apps/web/pages/reschedule/[uid].tsx

```diff
@@ -130,7 +138,7 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
       ? `team/${eventType.team.slug}`
       : dynamicEventSlugRef
       ? booking.dynamicGroupSlugRef
-      : booking.user?.username || "rick" /* This shouldn't happen */
+      : booking.user?.movedToProfile?.username || booking.user?.username || "rick" /* This shouldn't happen */
```

Comment:
It needs to be handled in a better way.

`movedToProfile.username` should always be accessible on org domain(like acme.cal.local) and `booking.user.username` should be accessed on non org domain(like app.cal.local).

So, earlier it seems that we were always accessing this route through `app.cal.com` due to which  it was always working.

But now, we have done a fix that causes `acme.cal.local` to be used for a reschedule of an org booking but there could be some existing links accessing the reschedule through app.cal.com which will get broken.


I think we should choose the redirect domain based on what we got.

So, if `user.movedToProfile.username` **is not nullish** and the request host is **not the org domain** then the redirect URL should be absolute using the org domain, else we just keep using the non-absolute URL for redirects. This is really important because `user.movedToProfile.username` only exists in org context which requires org domain.

Reply:
Made the changes as per the request.

Reply:
@kart1ka I found a few more issues with what I suggested and I am working on a different approach here 
https://github.com/calcom/cal.com/pull/15736

I will take care of fixing reschedule/[uid] endpoint as it has higher priority. 

I will take care of merging this PR after that.